### PR TITLE
get_vec for MatZ

### PR DESCRIPTION
**Pull request by Marcel583**
Implements  get_vec for MatZ.

### CODE REVIEW

File: src/integer/mat_z/get.rs [hunk 1]
```diff
@@ -252,6 +312,49 @@ mod test_get_num {
...
+#[cfg(test)]
+mod test_get_vec {
+
+    use crate::integer::MatZ;
+    use std::str::FromStr;
+
+    #[test]
+    fn get_vec_row_works() {
+        let matrix = MatZ::from_str(&format!("[[1,2,3],[4,{},{}]]", i64::MAX, i64::MIN)).unwrap();
+        let row = matrix.get_vec_row(1).unwrap();
```
Comment (jnsiemer) [inline src/integer/mat_z/get.rs:324]:
```suggestion
        let matrix = MatZ::from_str(&format!("[[0,0,0][1,2,3],[4,{},{}]]", i64::MAX, i64::MIN)).unwrap();
        let row_0 = matrix.get_row(0).unwrap();
        let row_1 = matrix.get_row(1).unwrap();
        let row_2 = matrix.get_row(2).unwrap();
```
Please test all rows that you initialized and also include a zero row/column to the tests and compare it appropriately. I shortly took a look whether a check if the row/column is zero is worth it, but I believe that the check takes up almost as much time as fetching and setting the entire row (and getting a zero vector should hopefully not be done too often).

**Review by jnsiemer — CHANGES_REQUESTED** [review]
Good work, I have some suggestions which would make the functions more efficient, the tests more meaningful, and the doc comments according to our specification.

File: src/integer/mat_z/get.rs [hunk 2]
```diff
@@ -96,6 +96,66 @@ impl GetEntry<Z> for MatZ {
...
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds)
+    /// if the number of the row is greater than the matrix or negative.
+    pub fn get_vec_row(&self, row: impl TryInto<i64> + Display + Copy) -> Result<Self, MathError> {
+        let mut out = MatZ::new(1, self.get_num_columns()).unwrap();
+        for column in 0..self.get_num_columns() {
+            out.set_entry(0, column, &self.get_entry(row, column)?)
+                .unwrap();
```
[the remark this answers is not in the document]
Comment (Marcel583) [reply src/integer/mat_z/get.rs]: I also need to implement evaluate indices manually for one index too

**Review by jnsiemer — APPROVED** [review]
Great work!

**Pull request merged**